pkg/resource/azurepathfixjob.go

```diff
@@ -95,6 +100,7 @@ func (gapfj *generatorAzurePathFixJob) expected() (runtime.Object, error) {
 		{Name: "AZURE_FEDERATED_TOKEN_FILE", Value: azureCfg.FederatedTokenFile},
 		{Name: "AZURE_ACCOUNTKEY", ValueFrom: &corev1.EnvVarSource{
 			SecretKeyRef: &corev1.SecretKeySelector{
+				Optional: &optional,
```

Comment:
This should fix the case when this secret is missing - apparently when the cluster is configured with workload identities this secret is missing (see bug report for details)